### DISCUSSION TITLE
some extra settings/options request(?)

### DISCUSSION THREAD

**Discussion by GyroJ** [d2phap/ImageGlass#816] [General]
Hi, I just installed the software and play around the option menu. Looks really nice to me. Great work!

I use the slideshow quite often when I want to have the image occupy my whole second monitor at work, so there're some suggestions. Kindly see if you like doing them later.

1. I prefer having the very image stay there in the slideshow mode forever until I finish writing something about it, so now I've set the timer to max 10mins. I can pause it ofc, but the setting doesn't stay when I re-enter the slideshow mode with another picture. If there's an option to set "pause" as default, that'd be great for my work flow.

2. I recognize that when entering the full-screen and slideshow mode, there's a small sentence pops up at the center reminding users how to exit the mode. It is great for newbies to prevent accidentally triggering the mode, but most will know that after like at most a few times, then it get a bit annoying to have this pop-up. I'd suggest adding an option to disable that small reminders if it's not already here.

3. I also have a (terrible) habit of using "Enter" as a shortcut to enter slideshow mode for image viewing. If there're more shortcut customization for power users (not me, I'm just a maniac), that'd be very nice.

Thanks.

**Comment by d2phap** [comment]
> 1... I can pause it ofc, but the setting doesn't stay when I re-enter the slideshow mode with another picture.

To sum up this, 
- Enter slideshow mode with 5s interval.
- Wait for few seconds, let's say the countdown timer is now 3s
- Pause slideshow
- View other image
- Resume slideshow
- The countdown timer should be 3s, instead of reset to 5s.

I may add an option to continue with the current countdown. 🤔


> 2...  I'd suggest adding an option to disable that small reminders if it's not already here...

You can uncheck the option "Show toast message". It will disable all the message of ImageGlass.
![image](https://user-images.githubusercontent.com/3154213/86536140-da8c4400-bf17-11ea-8736-c713f66d2b38.png)

> 3... using "Enter" as a shortcut to enter slideshow mode...

Currently, IG has limited keyboard customization, and using Enter to toggle slideshow mode is not supported.

**Reply by GyroJ** [reply]
> ...
> - Pause slideshow
> - View other image
> - Resume slideshow
> - The countdown timer should be 3s, instead of reset to 5s.

If it also saves the "pause slideshow" status, that'll be great.

Edit: I don't really use the countdown, so it'd be better if there's an option to set "pause slideshow" as default and the countdown only begins when I hit the Spacebar. Perhaps there can be like 3 options: (1) begin slideshow w/ a countdown; (2) begin slideshow w/ previous countdown status; and (3) begin slideshow w/o any countdown. I think the current countdown toggle works well with these options.

> You can uncheck the option "Show toast message". It will disable all the message of ImageGlass.

Thanks for the heads-up as I totally missed it!

> Currently, IG has limited keyboard customization, and using Enter to toggle slideshow mode is not supported.

That's unfortunate. Is there any ways right now to rebind keyboard outside the "Setting/Keyboard/" framework?

**Comment by d2phap** [comment]
> Is there any ways right now to rebind keyboard outside the "Setting/Keyboard/" framework?

Until it is implemented, but no plan for now.
